lib/semantics/check-omp-structure.cc

```diff
@@ -114,18 +113,57 @@ void OmpStructureChecker::Enter(const parser::OmpLoopDirective::Do &) {
   // reserve for nesting check
   SetContextDirectiveEnum(OmpDirective::DO);
 
-  OmpClauseSet allowed{OmpClause::PRIVATE, OmpClause::LASTPRIVATE,
-      OmpClause::LINEAR, OmpClause::REDUCTION, OmpClause::SCHEDULE,
-      OmpClause::COLLAPSE, OmpClause::ORDERED};
+  OmpClauseSet allowed{OmpClause::PRIVATE, OmpClause::FIRSTPRIVATE,
```

Comment:
I think you forgot to update the comment before this Enter() to match whatever is listed here.

Reply:
nice catch

Reply:
Addressed in commit [1b57193](https://github.com/flang-compiler/f18/pull/536/commits/1b57193d2377c5121f16075950b64ea77fa72e2e).